incubator/validator-core/src/main/java/io/aklivity/zilla/runtime/validator/core/StringValueValidator.java

```diff
@@ -18,14 +18,15 @@
 
 import org.agrona.DirectBuffer;
 
-import io.aklivity.zilla.runtime.engine.validator.Validator;
+import io.aklivity.zilla.runtime.engine.validator.ValueValidator;
+import io.aklivity.zilla.runtime.engine.validator.function.ValueConsumer;
 import io.aklivity.zilla.runtime.validator.core.config.StringValidatorConfig;
 
-public final class StringValidator implements Validator
+public final class StringValueValidator implements ValueValidator
 {
     private Predicate<byte[]> predicate;
 
-    public StringValidator(
+    public StringValueValidator(
```

Comment:
Define a static `ENCODING_VALIDATORS`map of string (encoding) to Predicate<byte[]> and make isValidXXX method static too.
Define a static catch-all case, `INVALID_ENCODING`, with lambda `b -> false`.
Then in constructor use `ENCODING_VALIDATORS.getOrDefault(config.encoding, INVALID_ENCODING)` to lookup correct predicate.